syntax = "proto3";

package test;

service UserStorage{
    rpc QueryUserInfo (CommonRequest) returns (QueryReply) ;
	rpc InsertUser    (CommonRequest) returns (CommonReply);
	rpc DeleteUser    (CommonRequest) returns (CommonReply);
	rpc UpdateUser    (CommonRequest) returns (CommonReply);
}


message User {
    string stu_name=1;
    int64 stu_id=2;
	int64 stu_age=3;
	string stu_sex=4;
}

message CommonReply{
    int64  code = 1;    // code = 0  表示正常返回
    string message = 2; // msg  = "" 表示正常返回
}

message QueryReply{
    repeated User user_list = 1;
}

message CommonRequest{
    User  cond = 1;
}
